src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerUdpClient.cs

```diff
@@ -45,31 +49,12 @@ public ValueTask<int> SendAsync(byte[] buffer, CancellationToken cancellationTok
 
         public ValueTask<int> SendAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken = default)
         {
-            var socket = this.client.Client;
-
-            var asyncResult = socket.BeginSend(
-                buffer,
-                offset,
-                count,
-                SocketFlags.None,
-                callback: null,
-                state: null);
-
-            if (asyncResult.CompletedSynchronously)
-            {
-                return new ValueTask<int>(socket.EndSend(asyncResult));
-            }
-
-            var tcs = new TaskCompletionSource<int>();
-
-            ThreadPool.RegisterWaitForSingleObject(
-                waitObject: asyncResult.AsyncWaitHandle,
-                callBack: (s, t) => tcs.SetResult(socket.EndSend(asyncResult)),
-                state: null,
-                millisecondsTimeOutInterval: -1,
-                executeOnlyOnce: true);
-
-            return new ValueTask<int>(tcs.Task);
+#if NETSTANDARD2_1
+            return this.client.Client.SendAsync(new ReadOnlyMemory<byte>(buffer, offset, count), SocketFlags.None, cancellationToken);
+#else
+            Debug.Assert(offset == 0, "Offset isn't supported in .NET Standard 2.0.");
```

Comment:
I don't think we need to include the `Debug.Assert` here - this isn't an error and we don't need to output it for every call to `SendAsync`.

Reply:
This relates to the `this.client.SendAsync(buffer, count)` below. It's not respecting the offset parameter, I couldn't find a way to do that in the .NET Standard 2.0 API. So if someone passes offset > 0 expecting the client to respect that, we'll be in trouble. The way the code works currently it will always be 0, but I thought the Assert would be a good way to prevent accidental misusage in the future. LMK if you still want me to remove.

Reply:
Debug.Assert only gets output when the library is compiled in debug builds. See MSDN [docs](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=netframework-4.8). Trace.Assert would generate the output for both debug and release builds and most of the time people will be using a release build via Nuget.

I'm not sure how I feel about it. This is an internal class and won't be called directly by a developer whom could pass 0 for the offset. I think an inline message here would be okay. We don't want to spam output to trace listeners.

@SergeyKanzhelev @cijothomas what are you feelings here?

Reply:
@MikeGoldsmith All your points are why I went with Debug.Assert :) I want it to blow up for the OT devs working on the lib (in Debug) so if someone makes a mistake we'll catch it immediately, but then we remove the check for release because the consumers don't need to worry about it.

Reply:
I see and understand :+1: